services/activitylog/pkg/service/service.go

```diff
@@ -13,25 +15,35 @@ import (
 	"github.com/cs3org/reva/v2/pkg/rgrpc/todo/pool"
 	"github.com/cs3org/reva/v2/pkg/storagespace"
 	"github.com/cs3org/reva/v2/pkg/utils"
+	"github.com/go-chi/chi/v5"
+	microstore "go-micro.dev/v4/store"
+
 	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	ehsvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/eventhistory/v0"
+	settingssvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/settings/v0"
 	"github.com/owncloud/ocis/v2/services/activitylog/pkg/config"
-	microstore "go-micro.dev/v4/store"
 )
 
-// Activity represents an activity
-type Activity struct {
+// RawActivity represents an activity as it is stored in the activitylog store
+type RawActivity struct {
 	EventID   string    `json:"event_id"`
 	Depth     int       `json:"depth"`
 	Timestamp time.Time `json:"timestamp"`
 }
 
 // ActivitylogService logs events per resource
 type ActivitylogService struct {
-	cfg    *config.Config
-	log    log.Logger
-	events <-chan events.Event
-	store  microstore.Store
-	gws    pool.Selectable[gateway.GatewayAPIClient]
+	cfg        *config.Config
+	log        log.Logger
+	events     <-chan events.Event
+	store      microstore.Store
+	gws        pool.Selectable[gateway.GatewayAPIClient]
+	mux        *chi.Mux
+	evHistory  ehsvc.EventHistoryService
+	valService settingssvc.ValueService
+	lock       sync.RWMutex
```

Comment:
It is a nitpick but could we use `mu sync.RWMutex` which is the de facto standard?